### DISCUSSION TITLE
What can trigger multiple simultaneous tries to install a parser with `auto_install=true`?

### DISCUSSION THREAD

**Discussion by msva** [nvim-treesitter/nvim-treesitter#7163] [Q&A] [answered]
Hi there!
I'm facing strange issue: when `auto_install` functionality triggers (say, when I open some file of type that treesitter parser is not installed for yet), I see it tries to install it multiple times in parallel (2 on my laptop and 3 on some servers).

Obviously, only one of them succeeds, and other fails with errors like
`"tree-sitter-%{lang}.tar.gz": no such file`, or the kinda-opposite
`tree-sitter-%{lang}-tmp`, or so on.

I guess it may be caused by auto_install hook installed multiple times somehow, but I don't know neither how to check it, nor how to fix.
Or may it be something else?

**Comment by msva** [comment]
by the way, I tried to add `vim.notify("ts init called")` in the config file, that is "attached" to nvim-treesitter `config` function in package manager (Lazy), and there is only one call.
But still 2 parallel tries to install a parser (I'm testing on awk)

**Reply by clason** [reply]
Don't lazy-load nvim-treesitter; that's not supported (and can trigger exactly that sort of race condition).

**Reply by msva** [reply]
Well,
1) okay, I will not, but
2) It behaves exactly the same with `lazy = false` :shrug:

**Reply by clason** [reply]
Then you might be opening multiple buffers at the same time? It's _very_ hard to say without a (minimal) reproducible setup. You asked in essence "what could be causing this?", and that's what my comment replied to.

**Reply by msva** [reply]
1) Nope, I'm only open (non-existent) `a.awk`, and facing this
2) currently I'm trying to build minimal reproducible config out of my dotfiles, but it seems it will not be as easy as I hope :-\

But I'll try to make something closer to it in near time

**Reply by msva** [reply]
(btw, is it any way to somehow debug it in-place (in the environment of my dotfiles), if any?)

**Reply by clason** [reply]
Sorry, no; the best (and only) way to debug such issues is to create a clean and minimal environment. 90% of issues arise from interactions between plugins and settings.

**Comment by msva** [comment, marked as answer]
While I prepared mini-repro config, I found that it was caused by nvim-colorizer (especially by attaching it's loading to `FileType` event).
Removing that fixed the issue with nvim-treesitter.

Thanks
